### PR TITLE
Update kafka vendor to support detect kafka broker controller

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
Test function for sending log couldn't support auto-detect kafka broker
controller now, when user input a broker endpoint which isn't the controller couldn't create kafka topic.

Solution:
update kafka vendor and detect broker controller

Issue:
https://github.com/rancher/rancher/issues/19448

Related test PR:
https://github.com/rancher/validation/pull/159

### CODE REVIEW

File: pkg/controllers/user/logging/utils/kafka.go [hunk 1]
```diff
@@ -49,91 +56,186 @@ func (w *kafkaTestWrap) TestReachable(dial dialer.Dialer, includeSendTestLog boo
...
+			return errors.Wrapf(err, "couldn't create topic %s", w.Topic)
+		}
+
+		if includeSendTestLog {
+			if err = w.sendData2Kafka(broker0Conn, dial); err != nil {
 				return err
 			}
 		}
+	} else {
+		brokerControllerTLSConfig, err := buildTLSConfig(w.Certificate, w.ClientCert, w.ClientKey, "", "", brokerController.Host, true)
```
Comment (gitlawr) [inline pkg/controllers/user/logging/utils/kafka.go]: Should we only build TLSConfig when  the schema matches `https` like L69 and formerly?

File: pkg/controllers/user/logging/utils/kafka.go [hunk 2]
```diff
@@ -13,13 +14,19 @@ import (
 )
 
 var (
-	kafkaTestData = kafka.Message{Value: []byte(testMessage)}
+	defaultPartition = 0
```
Comment (gitlawr) [inline pkg/controllers/user/logging/utils/kafka.go]: If we are using it as a constant, specify `const` for it, and the declaration in L183 is not needed.

**Comment by aiwantaozi** [comment]
updated

**Comment by gitlawr** [comment]
LGTM

**Pull request merged**